File: backend/src/main/java/nextpos/app/nextpos/service/interf/SaleService.java
package nextpos.app.nextpos.service.interf;

import nextpos.app.nextpos.model.dto.request.CreateRequest.CreateSaleRequest;
import nextpos.app.nextpos.model.dto.request.UpdateRequest.UpdateSaleRequest;
// import nextpos.app.nextpos.model.dto.response.RecentSaleResponse;
import nextpos.app.nextpos.model.dto.response.SaleResponse;

// import java.math.BigDecimal;
import java.util.List;
import java.util.Map;

public interface SaleService {
    SaleResponse createSale(CreateSaleRequest request);

    SaleResponse getSaleById(Long id);

    List<SaleResponse> getMySales();

    List<SaleResponse> getAllSales();

    SaleResponse updateSale(Long id, UpdateSaleRequest request);

    void deleteSale(Long id);

    // BigDecimal getTotalSales();

    // List<RecentSaleResponse> getRecentSales();

    List<SaleResponse> findRecentSalesByTenant(Long tenantId, int limit);

    Map<String, Object> getSalesSummary(String period, Long warehouseId);
}

File: frontend/src/services/saleService.ts
import api from "./api";
import type { Sale, CreateSaleRequest, UpdateSaleRequest } from "@/types/Sale";

export const getSales = () => api.get<Sale[]>("/sales");

export const getSaleById = (id: number) => api.get<Sale>(`/sales/${id}`);

export const getAllSales = () => api.get<Sale[]>("/sales/company");

export const createSale = (data: CreateSaleRequest) =>
  api.post<Sale>("/sales", data);

export const updateSale = (id: number, data: UpdateSaleRequest) =>
  api.put<Sale>(`/sales/${id}`, data);

export const deleteSale = (id: number) => api.delete<void>(`/sales/${id}`);